migrations/statictypes/statictype_migration.go

```diff
@@ -565,7 +569,8 @@ func CanSkipStaticTypeMigration(valueType interpreter.StaticType) bool {
 		return CanSkipStaticTypeMigration(valueType.ElementType())
 
 	case *interpreter.OptionalStaticType:
-		return CanSkipStaticTypeMigration(valueType.Type)
+		// optional types need to be rewritten as a result of changes to TypeID printing
+		return false
```

Comment:
Not sure we actually need this change. 

The skipping is based on the _value's type_, and an optional value does not have to be migrated.
Instead, we need to migrate values in which an optional static type appears, for example, in typed capability values (`Capability<Int?>`, capability static type), type values (`Type<Int?>()`, primitive static type meta type, etc.)

Reply:
@SupunS @dsainati1 I'm not 100% about the above, wdyt? If it makes sense, I'll revert the change

Reply:
makes sense to me. should be OK to revert.

Reply:
Yeah, same

Reply:
6c04cd4dc0740fc2d28e27cbe144d154aab5196c